common/src/main/java/com/wynntils/core/features/FeatureRegistry.java

```diff
@@ -4,66 +4,127 @@
-import com.wynntils.features.WynncraftButtonFeature;
+import com.wynntils.core.features.properties.EventListener;
+import com.wynntils.core.features.properties.KeyBinds;
+import com.wynntils.core.features.properties.StartEnabled;
+import com.wynntils.core.keybinds.KeyHolder;
 import com.wynntils.features.debug.ConnectionProgressFeature;
-import com.wynntils.features.debug.KeyBindTestFeature;
 import com.wynntils.features.debug.PacketDebuggerFeature;
+import com.wynntils.features.internal.FixPacketBugsFeature;
+import com.wynntils.features.internal.LootrunFeature;
+import com.wynntils.features.user.DialogueOptionOverrideFeature;
+import com.wynntils.features.user.EmeraldPouchHotkeyFeature;
+import com.wynntils.features.user.GammabrightFeature;
+import com.wynntils.features.user.HealthPotionBlockerFeature;
+import com.wynntils.features.user.IngredientPouchHotkeyFeature;
+import com.wynntils.features.user.ItemGuessFeature;
+import com.wynntils.features.user.ItemHighlightFeature;
+import com.wynntils.features.user.ItemScreenshotFeature;
+import com.wynntils.features.user.ItemStatInfoFeature;
+import com.wynntils.features.user.MountHorseHotkeyFeature;
+import com.wynntils.features.user.MythicBlockerFeature;
+import com.wynntils.features.user.PlayerGhostTransparencyFeature;
+import com.wynntils.features.user.SoulPointTimerFeature;
+import com.wynntils.features.user.WynncraftButtonFeature;
 import com.wynntils.mc.event.ClientTickEvent;
 import com.wynntils.mc.event.RenderEvent;
 import com.wynntils.mc.utils.CrashReportManager;
 import com.wynntils.mc.utils.McUtils;
 import com.wynntils.wc.utils.WynnUtils;
-import java.util.HashMap;
+import java.lang.reflect.Field;
+import java.util.ArrayList;
 import java.util.List;
-import java.util.Map;
 import net.minecraftforge.eventbus.api.SubscribeEvent;
 
 /** Loads {@link Feature}s */
 public class FeatureRegistry {
-    private static final Map<Class<? extends Feature>, Feature> FEATURES = new HashMap<>();
-    private static final Map<Class<? extends Overlay>, Overlay> OVERLAYS = new HashMap<>();
+    private static final List<Feature> FEATURES = new ArrayList<>();
+    private static final List<Overlay> OVERLAYS = new ArrayList<>();
 
     public static void registerFeature(Feature feature) {
         if (feature instanceof Overlay overlay) {
-            OVERLAYS.put(overlay.getClass(), overlay);
+            OVERLAYS.add(overlay);
         }
 
-        FEATURES.put(feature.getClass(), feature);
+        FEATURES.add(feature);
+
+        initializeFeature(feature);
+    }
+
+    private static void initializeFeature(Feature feature) {
+        Class<? extends Feature> featureClass = feature.getClass();
+
+        // instance field
+        try {
+            Field instanceField = featureClass.getDeclaredField("INSTANCE");
+            instanceField.set(null, feature);
+        } catch (Exception e) {
+            Reference.LOGGER.error("Failed to create instance object in " + featureClass.getName());
```

Comment:
This effectively forces all instances to exist, which ends up being kind of boilerplaty, though there likely lacks a better alternative.

Reply:
Also if this is called from the constructor it can prevent the existence of duplicate feature instances

Reply:
This couldn't be called from the constructor without duplicating code in every feature class, I believe.
It would be possible to make the instance field optional, but I don't think it matters much either way - we might as well enforce it for every feature imo

Reply:
Surely you could just call it from the parent constructor?

Reply:
it's a static field, so no - plus it wouldn't have the right type if you did that

unless you mean using reflection in the parent constructor, but i don't believe that would work either. you'd have to include `INSTANCE = this;` in the constructor of every feature class

Reply:
yeah

Reply:
I think it would be better to check if INSTANCE exists, and only initialize it if that is the case. Otherwise you'll force all features to have it, boilerplate as incomplete calls it, even if it's not used . My move with FeatureBase was to try to reduce such code to a minimum.

Reply:
that's a fair point - it's not like there's any extra work involved to make them optional, so i'll do that